### PR TITLE
Add more MagicWords for private, closed and active wikis, and add parser for {{NUMBEROFWIKISINCATEGORY}} (T6224)

### PR DESCRIPTION
**Pull request by Universal-Omega**

### CODE REVIEW

File: includes/WikiDiscover.php [hunk 1]
```diff
@@ -134,6 +134,21 @@ public static function onParserGetVariableValueSwitch(
 			global $wgLocalDatabases;
 			$ret = $cache[$magicWordId] = count( $wgLocalDatabases );
 		}
+		if ( $magicWordId == 'numberofprivatewikis' ) {
+			global $wgCreateWikiDatabase;
+			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+			$ret = $cache['numberofprivatewikis'] = $dbw->selectRowCount( 'cw_wikis', '*', 'wiki_private = 1' );
```
Comment (JohnFLewis) [inline includes/WikiDiscover.php:140]: Why not $magicWordId?

Comment (JohnFLewis) [reply includes/WikiDiscover.php:140]: Same for below, why are we not following standard of [ $table => $value ]?

Comment (Universal-Omega) [reply includes/WikiDiscover.php:140]: > Why not $magicWordId?

Oh, sorry that was unintentional when I was testing it I tested multiple ways and forgot to revert it back, I will fix that, thank you!

File: includes/WikiDiscover.php [hunk 2]
```diff
@@ -130,19 +130,53 @@ public static function onParserGetVariableValueSwitch(
...
+		if ( $magicWordId == 'numberofclosedwikis' ) {
+			global $wgCreateWikiDatabase;
+			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+			$ret = $cache[$magicWordId] = $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_closed' => 1 ] );
+		}
+		
 		return true;
 	}
 
+	public static function numberofwikisincategory( Parser $parser, $category = '' ) {
```
Comment (JohnFLewis) [inline includes/WikiDiscover.php:161]: camelCase + if we're defining the type for $parser, can we define the type of $category?

Comment (Universal-Omega) [reply includes/WikiDiscover.php]: Fixed

**Pull request merged**